torchdiffeq/_impl/rk_common.py

```diff
@@ -140,14 +156,34 @@ def __init__(self, func, y0, rtol, atol, first_step=None, safety=0.9, ifactor=10
         self.mid = self.mid.to(device=device, dtype=y0.dtype)
 
     def _before_integrate(self, t):
+        t0 = t[0]
         f0 = self.func(t[0], self.y0)
         if self.first_step is None:
             first_step = _select_initial_step(self.func, t[0], self.y0, self.order - 1, self.rtol, self.atol,
                                               self.norm, f0=f0)
         else:
             first_step = self.first_step
         self.rk_state = _RungeKuttaState(self.y0, f0, t[0], t[0], first_step, [self.y0] * 5)
-        self.next_grid_index = min(bisect.bisect(self.grid_points.tolist(), t[0]), len(self.grid_points) - 1)
+
+        # Handle step_t and jump_t arguments.
+        if self.step_t is None:
+            step_t = torch.tensor([], dtype=self.dtype, device=self.y0.device)
+        else:
+            step_t = _sort_tvals(self.step_t, t0)
+            step_t = step_t.to(self.dtype)
+        if self.jump_t is None:
+            jump_t = torch.tensor([], dtype=self.dtype, device=self.y0.device)
+        else:
+            jump_t = _sort_tvals(self.jump_t, t0)
```

Comment:
These calls to `_sort_tvals` have `O(n^2 log n)` complexity in the length `n` of the solve: each sort costs `O(n log n)` and these are called `O(n)` times. In contrast the original implementation was chosen to only cost `O(n)`.

If you want to not worry about order, then the sorting can be done up-front (rather than on every step!) and then the previous method used.

Reply:
This sort is only done once per odeint, so it should be okay. Maybe we can revisit this if it's shown to be a bottleneck.

Reply:
Ah sorry, this is in _before_integrate. Yup, agreed that's fine.